shared/utils/combine-remediations.py

```diff
@@ -284,20 +284,26 @@ def expand_xccdf_subs(fix, remediation_type, remediation_functions):
                     # This chunk contains call of 'populate' function
                     if "populate" in fixparts[idx]:
                         varname, fixtextcontribution = get_populate_replacement(remediation_type, fixparts[idx])
-                        # Append the contribution
-                        fix.text += fixtextcontribution
                         # Define new XCCDF <sub> element for the variable
-                        xccdfvarsub = ElementTree.Element("sub",
+                        xccdfvarsub = ElementTree.SubElement(fix, "sub",
                                                         idref=varname)
+
+                        # If this is first sub element,
+                        # the textcontribution needs to go to fix text
+                        # other wise, append to last subelement
+                        if list(fix).index(xccdfvarsub) == 0:
```

Comment:
This construct is kind of weird and wasteful. it materializes the entire iterable into a list and then looks whether the first element is something in the most wasteful way. Wouldn't it be easier to use `next(fix)` and compare directly to `xccdfvarsub`? the index gets reused later and instead of being stored it's calculated again.

Reply:
furthermore, you just appended the xccdfvarsub. there are way faster and more readable ways to check these two conditions. If the `fix` elements has no child elements then the first condition is met. Otherwise the new index will be the number of children. I think that's why this code avoided SubElement before.

Reply:
Updated to use number of children to identify first sub element.